packages/devtools_app/lib/src/performance/performance_controller.dart

```diff
@@ -315,16 +317,28 @@ class PerformanceController extends DisposableController
       return;
     }
 
-    if (!frame.isWellFormed &&
-        (_firstWellFormedFrameMicros == null ||
-            frame.timeFromFrameTiming.start.inMicroseconds >=
-                _firstWellFormedFrameMicros)) {
+    final bool frameBeforeFirstWellFormedFrame =
+        firstWellFormedFrameMicros == null ||
+            frame.timeFromFrameTiming.start.inMicroseconds <
+                firstWellFormedFrameMicros;
+    if (!frame.isWellFormed && !frameBeforeFirstWellFormedFrame) {
       // Only try to pull timeline events for frames that are after the first
       // well formed frame. Timeline events that occurred before this frame will
       // have already fallen out of the buffer.
       await processAvailableEvents();
     }
 
+    // If the frame is still not well formed after processing all available
+    // events, wait a short delay and try to process events again after the
+    // VM has been polled one more time.
+    if (!frame.isWellFormed && !frameBeforeFirstWellFormedFrame) {
+      _processing.value = true;
```

Comment:
we should ensure we are not already processing otherwise we might kick off multiple calls to processTraceEvents. Maybe that is fine but seems strange.